releasenotes/notes/rust-two-qubit-weyl-ec551f3f9c812124.yaml

```
@@ -0,0 +1,27 @@
+---
+features_synthesis:
+  - |
+    The :class:`.TwoQubitWeylDecomposition` synthesis class has been rewritten
+    in Rust for better performance.
+upgrade_synthesis:
+  - |
+    The :class:`.TwoQubitWeylDecomposition` no longer will self-specialize into
+    a subclass on creation. This was an internal detail of the :class:`.TwoQubitWeylDecomposition`
+    previously, and was not a documented public behavior as all the subclasses behaved
+    the same and were only used for internal dispatch. However, as it was discoverable
+    behavior this release note is to document that this will no longer occur and all
+    instances of :class:`.TwoQubitWeylDecomposition` will be of the same type. There is no
+    change in behavior for public methods of the class.
+
+deprecations_synthesis:
+  - |
+    The :meth:`.TwoQubitWeylDecomposition.specialize` method is now deprecated
+    and will be removed in the Qiskit 2.0.0 release. This method never had
+    a public purpose and was unsafe for an end user to call as it would
+    mutate the calculated decomposition in the object and produce invalid
+    fields in the object. It was only used internally to construct a new
+    :class:`.TwoQubitWeylDecomposition` object. Despite this it was still a
+    documented part of the public API for the class and is now being
+    deprecated without any potential replacement. This release it always will
+    raise a ``NotImplementedError` when called because the specialization
```

```suggestion
    raise a ``NotImplementedError`` when called because the specialization
```

(what the docs build is complaining about)

---

Fixed in: https://github.com/Qiskit/qiskit/pull/11946/commits/160d70552fcc4a3a29f5d3801849b82b6bebdb20